gcloud/bigquery/table.py

```diff
@@ -231,6 +231,59 @@ def table_type(self):
         return self._properties.get('type')
 
     @property
+    def partitioning_type(self):
+        """Time partitioning of the table.
+        :rtype: str, or ``NoneType``
+        :returns: Returns type if the table is partitioned, None otherwise.
+        """
+        partitioned = None
+        if "timePartitioning" in self._properties:
+            partitioned = self._properties.get('timePartitioning').get('type')
+        return partitioned
+
+    @partitioning_type.setter
+    def partitioning_type(self, value):
+        """Update the partitioning type of the table
+
+        :type value: str
+        :param value: partitioning type only "DAY" is currently supported
+        """
+        if not isinstance(value, six.string_types) or value.upper() != "DAY":
+            raise ValueError("value must be one of ['DAY']")
+        try:
+            self._properties['timePartitioning']['type'] = value.upper()
+        except KeyError:
+            self._properties['timePartitioning'] = {}
+            self._properties['timePartitioning']['type'] = value.upper()
```

Comment:
Rather than the `try: ... except:`, it would be simpler to use `setdefault`:

```python
        self._properties.setdefault('timePartitioning', {})['type'] = value.upper()
```